app/main/posts/modify/post-data-editor.controller.js

```diff
@@ -72,6 +74,12 @@ function PostDataEditorController(
     $scope.submitting = $translate.instant('app.submitting');
     $scope.hasPermission = $rootScope.hasPermission('Manage Posts');
     $scope.selectForm = selectForm;
+    $scope.isSaving = LoadingProgress.getSavingState();
+
+    LoadingProgress.subscribeOnSavingState(function (isSaving) {
```

Comment:
This is tricky... it adds a watch, which will end up hanging around after moving to another view. We either need to unbind the watch on $scope destroy - kinda messy again. Or the easier method would be to call `getSavingState` direct ie.
`$scope.isSaving = LoadingProgress.getSavingState`
Then use `isSaving()` instead of `isSaving` in the templates. I think angular still does a watch behind the scenes but it happens automatically

Reply:
Good idea! And it is working :) :) Will change it.